File: 169-majority-element/169-majority-element.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var majorityElement = function(nums) {
    let map = {}
    for(let i=0; i < nums.length; i++){
        map[nums[i]] = (map[nums[i]] || 0) + 1
      //  console.log(map[nums[i]])
        if(map[nums[i]] > nums.length/2) return nums[i]
    }
};

File: 169-majority-element/169-majority-element.java
class Solution {
    public int majorityElement(int[] nums) {
        if(nums.length == 1)
            return nums[0];
        int count = 0, candidate = 0;
        for(int i = 0; i < nums.length ; i++) {
            if(count == 0)
                candidate = nums[i];
            count = candidate == nums[i] ? ++count : --count;
        }
        return candidate;
    }
}